### ISSUE TITLE
Mouse detected by cli, not listed by UI

### ISSUE THREAD

**Issue by jeblad** [pwr-Solaar/Solaar#866]
**Information**

It seems like my Logitech M590 is correctly identified, but then dropped from the list in the UI, while still being detected by the cli. This was first observed on a Ubuntu 19.10, but later verified on Ubuntu 20.04. I'm not dumping 20.04 as it seems identical to 19.10.

**Ubuntu 19.10**

```
john@hydra:~$ solaar --version
solaar 1.0.1
```

```
john@hydra:~$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 19.10
Release:	19.10
Codename:	eoan
```

```
john@hydra:~$ uname -srmo
Linux 5.3.0-62-generic x86_64 GNU/Linux
```

```
john@hydra:~$ solaar show
Unifying Receiver
  Device path  : /dev/hidraw1
  USB id       : 046d:c52b
  Serial       : BC7A93C7
    Firmware   : 12.10.B0032
    Bootloader : 02.15
    Other      : AA.AA
  Has 2 paired device(s) out of a maximum of 6.
  Notifications: wireless (0x000100)
  Device activity counters: 1=33, 3=252

  1: M585/M590 Multi-Device Mouse
     Codename     : M585/M590
     Kind         : mouse
     Wireless PID : 406B
     Protocol     : HID++ 4.5
     Polling rate : 8 ms (125Hz)
     Serial number: 35ECB06B
        Bootloader: BOT 48.00.B0001
          Firmware: MPM 05.01.B0006
             Other: 
     The power switch is located on the base.
     Supports 33 HID++ 2.0 features:
         0: ROOT                   {0000}   
         1: FEATURE SET            {0001}   
         2: DEVICE FW VERSION      {0003}   
         3: DEVICE NAME            {0005}   
         4: WIRELESS DEVICE STATUS {1D4B}   
         5: RESET                  {0020}   
         6: unknown:0007           {0007}   
         7: BATTERY STATUS         {1000}   
         8: CHANGE HOST            {1814}   
         9: unknown:1815           {1815}   
        10: REPROG CONTROLS V4     {1B04}   
        11: unknown:1C00           {1C00}   
        12: POINTER SPEED          {2205}   
            Pointer Speed: 1.0
        13: VERTICAL SCROLLING     {2100}   
            Roller type: standard
            Ratchet per turn: 18
            Scroll lines: 0
        14: unknown:00C2           {00C2}   
        15: unknown:1802           {1802}   internal, hidden
        16: unknown:1803           {1803}   internal, hidden
        17: unknown:1806           {1806}   internal, hidden
        18: unknown:1805           {1805}   internal, hidden
        19: unknown:1813           {1813}   internal, hidden
        20: unknown:1830           {1830}   internal, hidden
        21: unknown:1861           {1861}   internal, hidden
        22: unknown:1890           {1890}   internal, hidden
        23: unknown:1891           {1891}   internal, hidden
        24: unknown:18A1           {18A1}   internal, hidden
        25: unknown:1DF3           {1DF3}   internal, hidden
        26: unknown:1E00           {1E00}   hidden
        27: unknown:1EB0           {1EB0}   internal, hidden
        28: unknown:18B1           {18B1}   internal, hidden
        29: unknown:1850           {1850}   internal, hidden
        30: unknown:1F03           {1F03}   internal, hidden
        31: unknown:18C0           {18C0}   internal, hidden
        32: LOWRES WHEEL           {2130}   
            Wheel Reports: HID
     Has 8 reprogrammable keys:
         0: LEFT CLICK                , default: LeftClick                   => LEFT CLICK                
             mse, reprogrammable, pos:0, group:1, gmask:1
         1: RIGHT CLICK               , default: RightClick                  => RIGHT CLICK               
             mse, reprogrammable, pos:0, group:1, gmask:1
         2: MIDDLE BUTTON             , default: MiddleMouseButton           => MIDDLE BUTTON             
             divertable, mse, persistently divertable, reprogrammable, pos:0, group:2, gmask:3
         3: BACK AS BUTTON 4          , default: BackEx                      => BACK AS BUTTON 4          
             divertable, mse, persistently divertable, reprogrammable, pos:0, group:2, gmask:3
         4: FORWARD AS BUTTON 5       , default: BrowserForwardEx            => FORWARD AS BUTTON 5       
             divertable, mse, persistently divertable, reprogrammable, pos:0, group:2, gmask:3
         5: LEFT SCROLL AS AC PAN     , default: HorzScrollLeftSet           => LEFT SCROLL AS AC PAN     
             divertable, mse, persistently divertable, reprogrammable, pos:0, group:2, gmask:3
         6: RIGHT SCROLL AS AC PAN    , default: HorzScrollRightSet          => RIGHT SCROLL AS AC PAN    
             divertable, mse, persistently divertable, reprogrammable, pos:0, group:2, gmask:3
         7: unknown:00D7              , default: unknown:00B4                => unknown:00D7              
             divertable, virtual, pos:0, group:3, gmask:0
     Battery: 50%, discharging.

  3: MX Keys Wireless Keyboard
     Codename     : MX Keys
     Kind         : keyboard
     Wireless PID : 408A
     Protocol     : HID++ 4.5
     Polling rate : 20 ms (50Hz)
     Serial number: 98242B65
        Bootloader: BL1 08.00.B0011
          Firmware: MPK 12.00.B0011
             Other: 
     The power switch is located on the edge of top right corner.
     Supports 33 HID++ 2.0 features:
         0: ROOT                   {0000}   
         1: FEATURE SET            {0001}   
         2: DEVICE FW VERSION      {0003}   
         3: DEVICE NAME            {0005}   
         4: WIRELESS DEVICE STATUS {1D4B}   
         5: RESET                  {0020}   
         6: unknown:0007           {0007}   
         7: BATTERY STATUS         {1000}   
         8: REPROG CONTROLS V4     {1B04}   
         9: CHANGE HOST            {1814}   
        10: unknown:1815           {1815}   
        11: unknown:1982           {1982}   
        12: K375S FN INVERSION     {40A3}   
        13: ENCRYPTION             {4100}   
        14: LOCK KEY STATE         {4220}   
        15: KEYBOARD DISABLE       {4521}   
        16: unknown:4531           {4531}   
        17: unknown:00C2           {00C2}   
        18: unknown:1802           {1802}   internal, hidden
        19: unknown:1803           {1803}   internal, hidden
        20: unknown:1806           {1806}   internal, hidden
        21: unknown:1813           {1813}   internal, hidden
        22: unknown:1805           {1805}   internal, hidden
        23: unknown:1830           {1830}   internal, hidden
        24: unknown:1890           {1890}   internal, hidden
        25: unknown:1891           {1891}   internal, hidden
        26: unknown:18A1           {18A1}   internal, hidden
        27: unknown:1DF3           {1DF3}   internal, hidden
        28: unknown:1E00           {1E00}   hidden
        29: unknown:1EB0           {1EB0}   internal, hidden
        30: unknown:1861           {1861}   internal, hidden
        31: unknown:1A20           {1A20}   internal, hidden
        32: unknown:18B0           {18B0}   internal, hidden
     Has 24 reprogrammable keys:
         0: unknown:00D1              , default: unknown:00AE                => unknown:00D1              
             divertable, nonstandard, persistently divertable, pos:0, group:0, gmask:0
         1: unknown:00D2              , default: unknown:00AF                => unknown:00D2              
             divertable, nonstandard, persistently divertable, pos:0, group:0, gmask:0
         2: unknown:00D3              , default: unknown:00B0                => unknown:00D3              
             divertable, nonstandard, persistently divertable, pos:0, group:0, gmask:0
         3: unknown:00C7              , default: unknown:00A3                => unknown:00C7              
             divertable, is FN, FN sensitive, persistently divertable, reprogrammable, pos:1, group:0, gmask:0
         4: unknown:00C8              , default: unknown:00A4                => unknown:00C8              
             divertable, is FN, FN sensitive, persistently divertable, reprogrammable, pos:2, group:0, gmask:0
         5: unknown:00E0              , default: unknown:00BF                => unknown:00E0              
             divertable, is FN, FN sensitive, persistently divertable, reprogrammable, pos:3, group:0, gmask:0
         6: unknown:00E1              , default: unknown:00C0                => unknown:00E1              
             divertable, is FN, FN sensitive, persistently divertable, reprogrammable, pos:4, group:0, gmask:0
         7: SHOW DESKTOP              , default: ShowDesktop                 => SHOW DESKTOP              
             divertable, is FN, FN sensitive, persistently divertable, reprogrammable, pos:5, group:0, gmask:0
         8: unknown:00E2              , default: unknown:00C1                => unknown:00E2              
             divertable, is FN, FN sensitive, persistently divertable, reprogrammable, pos:6, group:0, gmask:0
         9: unknown:00E3              , default: unknown:00C2                => unknown:00E3              
             divertable, is FN, FN sensitive, persistently divertable, reprogrammable, pos:7, group:0, gmask:0
        10: unknown:00E4              , default: Previous                    => unknown:00E4              
             divertable, is FN, FN sensitive, persistently divertable, reprogrammable, pos:8, group:0, gmask:0
        11: unknown:00E5              , default: Play/Pause                  => unknown:00E5              
             divertable, is FN, FN sensitive, persistently divertable, reprogrammable, pos:9, group:0, gmask:0
        12: unknown:00E6              , default: Next                        => unknown:00E6              
             divertable, is FN, FN sensitive, persistently divertable, reprogrammable, pos:10, group:0, gmask:0
        13: unknown:00E7              , default: Mute                        => unknown:00E7              
             divertable, is FN, FN sensitive, persistently divertable, reprogrammable, pos:11, group:0, gmask:0
        14: unknown:00E8              , default: Volume Down                 => unknown:00E8              
             divertable, is FN, FN sensitive, persistently divertable, reprogrammable, pos:12, group:0, gmask:0
        15: unknown:00E9              , default: Volume Up                   => unknown:00E9              
             divertable, nonstandard, persistently divertable, reprogrammable, pos:0, group:0, gmask:0
        16: Calculator                , default: Calculator                  => Calculator                
             divertable, nonstandard, persistently divertable, reprogrammable, pos:0, group:0, gmask:0
        17: unknown:00BF              , default: unknown:009B                => unknown:00BF              
             divertable, nonstandard, persistently divertable, reprogrammable, pos:0, group:0, gmask:0
        18: unknown:00EA              , default: unknown:00C3                => unknown:00EA              
             divertable, nonstandard, persistently divertable, reprogrammable, pos:0, group:0, gmask:0
        19: Lock PC                   , default: WindowsLock                 => Lock PC                   
             divertable, nonstandard, persistently divertable, reprogrammable, pos:0, group:0, gmask:0
        20: unknown:00EC              , default: unknown:00B8                => unknown:00EC              
             divertable, nonstandard, persistently divertable, pos:0, group:0, gmask:0
        21: unknown:00EB              , default: unknown:00B6                => unknown:00EB              
             divertable, nonstandard, persistently divertable, pos:0, group:0, gmask:0
        22: unknown:00DE              , default: Do Nothing One              => unknown:00DE              
             is FN, pos:0, group:0, gmask:0
        23: unknown:0034              , default: Do Nothing One              => unknown:0034              
             nonstandard, pos:0, group:0, gmask:0
     Battery: 50%, discharging.

```

**Describe the bug**
The M590 (mouse) is initially listed in the UI, then goes away.

It is always the M590 (mouse) that disappears from the list, never the MXkeys (keyboard).

**To Reproduce**
1. Do a fresh install (an old one probably work to)
2. Set up MX keys (keyboard) and M590 (mouse) on same unifying receiver
3. Reboot (or quit and restart Gnome Solaar UI)
4. Now M590 is gone from the UI listing

5. Double check with `solaar show` and the M590 (mouse) is still there.

**Screenshots**
Should be obvious, but if necessary I can provide screenshots.

**Additional context**
Both MX keys and M590 is on the same unifying receiver, and there are often but not always two machines, where one of them dual boots into Win10. Bug report are for both running Ubuntu.

**Comment by pfps** [comment]
Solaar 1.0.1 is getting old now, please update to a newer version.

Please run solaar as `solaar -dd` and paste the output of a run where the M590 is connected but does not show up in the UI.  There probably was some sort of exception that caused the device to not be listed.

**Comment by jeblad** [comment]
Note that solaar 1.0.1 is the current version in the repo for Ubuntu 20.04 LTS. Anyhow I can test a newer version.

**Comment by pfps** [comment]
Yes, Ubuntu generally has old versions of software.  If there is a problem, it's only going to be fixed if it is a problem in the current version. Installing a recent version is easy - `pip install --user solaar=1.0.3rc1` should work fine right now.

**Comment by jeblad** [comment]
Tested Ubuntu 19.10 and 20.04 with 1.0.2, and it works properly. At least I can't see any immediate problems.

Will test 1.0.3rc1.

**Comment by pfps** [comment]
If 1.0.2 works fine, then no *need* to test 1.0.3rc1 but this will be the latest version in a week so getting it tested would be very nice.

As this appears to be fixed in 1.0.2 I'm closing the issue.  If any problems show up later, just reopen it.

**Comment by jeblad** [comment]
Version 1.0.3rc1 seems to work as expected in both 19.10 and 20.04.
